src/wallet/scriptpubkeyman.h

```diff
@@ -0,0 +1,479 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_WALLET_SCRIPTPUBKEYMAN_H
+#define BITCOIN_WALLET_SCRIPTPUBKEYMAN_H
+
+#include <script/signingprovider.h>
+#include <script/standard.h>
+#include <wallet/crypter.h>
+#include <wallet/ismine.h>
+#include <wallet/walletdb.h>
+#include <wallet/walletutil.h>
+
+#include <boost/signals2/signal.hpp>
+
+enum class OutputType;
+
+// Wallet storage things that ScriptPubKeyMans need in order to be able to store things to the wallet database
+class WalletStorage
+{
+public:
+    virtual ~WalletStorage() = default;
+    virtual const std::string GetDisplayName() const = 0;
+    virtual std::shared_ptr<WalletDatabase> GetDatabase() = 0;
+    virtual bool IsWalletFlagSet(uint64_t) const = 0;
+    virtual void SetWalletFlag(uint64_t) = 0;
```

Comment:
SetWalletFlag method and override don't seem to be called anymore could be removed.

Reply:
It's used in descriptor wallets.